cc/crypto/hpke/jni/context_jni.cc

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2023 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "../recipient_context.h"
+#include "../sender_context.h"
+#include "com_google_oak_crypto_hpke_Context_RecipientRequestContext.h"
+#include "com_google_oak_crypto_hpke_Context_RecipientResponseContext.h"
+#include "com_google_oak_crypto_hpke_Context_SenderRequestContext.h"
+#include "com_google_oak_crypto_hpke_Context_SenderResponseContext.h"
+
+JNIEXPORT jbyteArray JNICALL
+Java_com_google_oak_crypto_hpke_Context_00024SenderRequestContext_nativeSeal(
+    JNIEnv* env, jobject obj, jbyteArray plaintext, jbyteArray associated_data) {
+  if (plaintext == NULL || associated_data == NULL) {
+    return {};
+  }
+
+  int plaintext_len = env->GetArrayLength(plaintext);
+  char* plaintext_buf = new char[plaintext_len];
```

Comment:
Is there a memory leak further down, when exiting via line 46? How about wrapping all `new` around here into unique_ptr? Then all delete[] lines can be removed, and safer as well.

Reply:
done.